packages/ingest/src/recursion.ts

```diff
@@ -1,7 +1,12 @@
-'use strict';
+import path from 'path';
+import * as log from '@cumulus/common/log';
 
-const path = require('path');
-const log = require('@cumulus/common/log');
+export interface RecursionFile {
+  name: string,
+  type: string
+}
+
+export type RecursionFunction = (x: string) => RecursionFile[];
```

Comment:
What is `x`? Can we give it a better name?

Reply:
https://github.com/nasa/cumulus/pull/1798/commits/08ceba70690ea61f11cecfef5c6a71d1e1c13c47#diff-fc77b0b42462304bc512aca0c3380f33R9